pyproject.toml

```diff
@@ -0,0 +1,42 @@
+[tool.poetry]
+name = "cantusdb"
+package-mode = false
+
+[tool.poetry.dependencies]
+python = "^3.9"
+Django = "4.2.11"
+django-autocomplete-light = "3.9.4"
+django-extra-views = "0.13.0"
+django-quill-editor = "0.1.40"
+django-reversion = "5.0.8"
+gunicorn = "22.0.0"
+psycopg = "3.1.8"
+requests = "2.31.0"
+ujson = "5.9.0"
+volpiano-display-utilities = { git = "https://github.com/DDMAL/volpiano-display-utilities.git", tag = "v1.1.2" }
+
+[tool.poetry.group.debug]
+optional = true
+
+[tool.poetry.group.debug.dependencies]
+django_debug_toolbar = "3.8.1"
+
+[tool.poetry.group.test]
+optional = true
+
+[tool.poetry.group.test.dependencies]
```

Comment:
I think I see where you're going with this, but wouldn't it be OK to just have `dev`? Do you routinely install the dev dependencies without the test ones? Or vice versa?

Reply:
I was waffling on whether to have separate `debug` and `test` groups, because I can't think of a case where you'd install one or not the other.

Technically, what's in the `dev` group currently doesn't ever need to be installed inside the container (code formatting and type checking can happen outside the container environment), whereas those dependencies that are in the `debug` and `test` groups do (because they extend, a la `django-debug-toolbar`, the django app itself).

So, it's a Docker-induced distinction. Does it really matter that we keep them separate? (ie. does anything break if we start installing black inside the container?) No, so we could definitely combine them.

Thoughts?